addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/FetchMetadataRequestHeadersScanRuleTest.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.network.HttpMessage;
+
+class FetchMetadataRequestHeadersScanRuleTest
+        extends PassiveScannerTest<FetchMetadataRequestHeadersScanRule> {
+    @Test
+    void shouldRaiseSfsAlertGivenRequestDoesntSendSfsHeader() throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader(
+                "GET / HTTP/1.1\r\n"
+                        + "Sec-Fetch-Mode: navigate\r\n"
+                        + "Sec-Fetch-Dest: audio\r\n"
+                        + "Sec-Fetch-User: ?1\r\n");
```

Comment:
This seems largely static through the majority of the tests, perhaps extract a method that allows customizations to be passed.

Maybe multiple methods.

.... whatever makes sense.

Reply:
Done. I have added 2 separate methods for missing and invalid cases respectively.